crates/distribution-types/src/resolution.rs

```diff
@@ -56,99 +57,76 @@ impl Resolution {
+                editable: None,
             },
-            Dist::Source(SourceDist::DirectUrl(sdist)) => Self {
-                name: sdist.name,
-                extras: vec![],
-                version_or_url: Some(pep508_rs::VersionOrUrl::Url(sdist.url)),
-                marker: None,
+            Dist::Source(SourceDist::Registry(sdist)) => UvSource::Registry {
+                version: pep440_rs::VersionSpecifiers::from(
+                    pep440_rs::VersionSpecifier::equals_version(sdist.filename.version.clone()),
+                ),
+                index: None,
             },
-            Dist::Source(SourceDist::Git(sdist)) => Self {
-                name: sdist.name,
-                extras: vec![],
-                version_or_url: Some(pep508_rs::VersionOrUrl::Url(sdist.url)),
-                marker: None,
+            Dist::Source(SourceDist::DirectUrl(sdist)) => UvSource::Url {
+                url: sdist.url.clone(),
+                subdirectory: None,
             },
-            Dist::Source(SourceDist::Path(sdist)) => Self {
-                name: sdist.name,
-                extras: vec![],
-                version_or_url: Some(pep508_rs::VersionOrUrl::Url(sdist.url)),
-                marker: None,
+            Dist::Source(SourceDist::Git(sdist)) => {
+                let git_url = ParsedGitUrl::try_from(sdist.url.raw())
+                    .expect("urls must be valid at this point");
+                UvSource::Git {
+                    url: sdist.url.clone(),
+                    repository: git_url.url.repository().clone(),
+                    reference: git_url.url.reference().clone(),
+                    subdirectory: git_url.subdirectory,
+                }
+            }
+            Dist::Source(SourceDist::Path(sdist)) => UvSource::Path {
+                path: sdist.path.clone(),
+                url: sdist.url.clone(),
+                editable: None,
             },
-        }
-    }
-}
-
-impl From<InstalledDist> for Requirement {
-    fn from(dist: InstalledDist) -> Self {
-        Self {
-            name: dist.name().clone(),
-            extras: vec![],
-            version_or_url: Some(pep508_rs::VersionOrUrl::VersionSpecifier(
-                pep440_rs::VersionSpecifiers::from(pep440_rs::VersionSpecifier::equals_version(
-                    dist.version().clone(),
-                )),
-            )),
-            marker: None,
-        }
-    }
-}
-
-impl From<ResolvedDist> for Requirement {
-    fn from(dist: ResolvedDist) -> Self {
-        match dist {
-            ResolvedDist::Installable(dist) => dist.into(),
-            ResolvedDist::Installed(dist) => dist.into(),
-        }
+        },
+        ResolvedDist::Installed(dist) => UvSource::Registry {
+            version: pep440_rs::VersionSpecifiers::from(
+                pep440_rs::VersionSpecifier::equals_version(dist.version().clone()),
+            ),
+            index: None,
+        },
+    };
+    UvRequirement {
```

Comment:
I think we should just name this `Requirement`, and use `pep508::Requirement` everywhere else, personally. I'd like to avoid `Uv` in the name of abstractions.

Reply:
(I'd apply this same judgment to all the other `Uv`-prefixed types. What do you think?)